hil-test/tests/i2c.rs

```diff
@@ -40,9 +39,9 @@ mod tests {
 
         // Create a new peripheral object with the described wiring and standard
         // I2C clock speed:
-        let i2c = I2c::new(peripherals.I2C0, sda, scl, 100.kHz());
-
-        Context { i2c }
+        Context {
+            i2c: I2c::new(peripherals.I2C0).with_sda(sda).with_scl(scl),
```

Comment:
Just a general observation, which applies mostly to i2c, but I guess some other drivers:

I suppose using the `with` pattern with i2c might end up with someone forgetting to call them... but tbh I think we need to expect some kind of embedded knowledge to know about sda/scl. That along with docs and examples should be enough to not worry about this I think.

Reply:
I've never been able to remember which parameter belongs to which pin. Yeah it's possible to forget calling these but now we are also able to change pins mid-operation, which allows third parties to write drivers that can change pins on a shared bus, using SetConfig.

Reply:
> which allows third parties to write drivers that can change pins on a shared bus, using SetConfig.

Not quite I don't think, because `with_` takes `self` and `SetConfig` gives `&mut self` - but we could change that if we wanted to. I see your point, and I do agree that overall `with_X` is a nice pattern.

Reply:
I'm fairly sure `*self = self.with_sda` works, although I'm not 100% sure.

Reply:
Nevermind it doesn't by itself (but the replace_with crate exists)